Modules/Scripted/EditorLib/HelperBox.py

```diff
@@ -165,19 +165,17 @@ def setVolumes(self,masterVolume,mergeVolume):
 
   def setMasterVolume(self,masterVolume):
     """select merge volume"""
+    if isinstance(masterVolume, basestring):
+      masterVolume = slicer.mrmlScene.GetNodeByID(masterVolume)
     self.masterSelector.setCurrentNode( masterVolume )
-    self.select()
```

Comment:
In general it is explicitly created on enter() - at least the user is
prompted to create one if one isn't found.

https://github.com/Slicer/Slicer/blob/master/Modules/Scripted/Editor/Editor.py#L131-L151

or did you mean something else?

On Wed, Jun 10, 2015 at 8:58 AM, Jean-Christophe Fillion-Robin <
notifications@github.com> wrote:

> In Modules/Scripted/EditorLib/HelperBox.py
> <https://github.com/Slicer/Slicer/pull/287#discussion_r32113948>:
>
> >      self.masterSelector.setCurrentNode( masterVolume )
> > -    self.select()
>
> Agreed.
>
> On a similar topic, I wonder if it would make more sense to explicitly
> created the label map if none is available instead of trying to lazily
> create it. It means that we would check if a merge volume is needed in the
> Editor/enter() function.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Slicer/Slicer/pull/287/files#r32113948>.
>